OpenSim/Region/CoreModules/Scripting/LSLHttp/UrlModule.cs

```diff
@@ -518,11 +518,16 @@ public void AsyncHttpRequest(IHttpServer server, string path, OSHttpRequest http
                     }
                 }
 
+                // Grab the raw unprocessed original query string, if any.
+                int rawQueryPos = httpRequest.Url.Query.IndexOf('?');
+                string rawQueryStr = (rawQueryPos < 0) ? httpRequest.Url.Query : httpRequest.Url.Query.Substring(rawQueryPos + 1);
+
                 //if this machine is behind DNAT/port forwarding, currently this is being
                 //set to address of port forwarding router
                 requestData.headers["x-remote-ip"] = httpRequest.RemoteIPEndPoint.ToString();
                 requestData.headers["x-path-info"] = pathInfo;
-                requestData.headers["x-query-string"] = queryString;
+                requestData.headers["x-query-string"] = rawQueryStr;        // raw original (SL-compatible)
+                requestData.headers["x-query-string-compat"] = queryString; // processed (old Halcyon scripts)
```

Comment:
That's exactly the reason for adding it.  I'm not sure it will help but it won't hurt to have a compatible code option available should someone need it.

We don't really have any other option here for the normal x-query-string implementation. It's just plain broken in many ways and needs to be fixed, and the fixes produce different results than the old code. So I've added a second option to return the former value. That would require a script update for anything that _relied_ on the old results, although that was a null string anyway, for many of the broken cases.  But anything that expected only an `abc` entry for `abc=def&ghi` at least has an easy method to get what the previous server would have returned.  Whether that will be used or not, I don't know. Just trying to cooperate with the old script logic as much as possible, just in case.